pkg/controller/deployment/util/deployment_util.go

```diff
@@ -261,14 +261,25 @@ func SetNewReplicaSetAnnotations(deployment *apps.Deployment, newRS *apps.Replic
 	// If a revision annotation already existed and this replica set was updated with a new revision
 	// then that means we are rolling back to this replica set. We need to preserve the old revisions
 	// for historical information.
-	if ok && annotationChanged {
+	if ok && oldRevisionInt < newRevisionInt {
```

Comment:
Can we assume that this number never wraps around and never goes backwards? I feel like this check should just be `!=` for pure deduplication since it doesn't seem like this function's job should to enforce this invariant.

Reply:
The condition also appears on line 256 (preexisting code).
Both are int64.